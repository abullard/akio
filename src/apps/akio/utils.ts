import fs from 'fs';
import path from 'path';
import { glob } from 'glob';

export const getPkgManager = () => {
    if (fs.existsSync('pnpm-lock.yaml')) return 'pnpm';
    if (fs.existsSync('yarn.lock')) return 'yarn';

    return 'npm';
};

const loadJson = async (path: string) => await import(path, { with: { type: 'json' } });

export const readAllPkgJsons = async () => {
    const paths = await glob('**/package.json', {
        signal: AbortSignal.timeout(1000),
        ignore: 'node_modules/**',
        absolute: true
    });

    const allPkgJsonContents = await Promise.all(paths.map(loadJson));
    for (const contents in allPkgJsonContents) {
        
    }

    // original impl
    return;
}
