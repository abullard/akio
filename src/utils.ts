import fs from 'fs';
import path from 'path';

export const getPkgManager = () => {
    if (fs.existsSync('pnpm-lock.yaml')) return 'pnpm';
    if (fs.existsSync('yarn.lock')) return 'yarn';

    return 'npm';
};

export const readPackageJson = () => {
    const pkgPath = path.resolve(process.cwd(), 'package.json');
    return JSON.parse(fs.readFileSync(pkgPath, 'utf8'));
}
