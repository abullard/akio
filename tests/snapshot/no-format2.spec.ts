import { describe, it, expect } from 'vitest';
import { spawnWrapper } from '../utils/test-utils';

describe('no-format2.spec.ts', () => {
    it('should turn formatting off, --format', async () => {
        const stubbedCommandForNoOp = 'test:stub';
        const skipInput = '-i';
        const cmd = 'pnpm';
        const args = ['akio', '--format', skipInput, stubbedCommandForNoOp];

        const { stdout } = await spawnWrapper(cmd, args);

        expect(stdout).toMatchSnapshot();
    });
});
