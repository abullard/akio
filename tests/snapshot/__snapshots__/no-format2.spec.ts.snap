// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-format2.spec.ts > should turn formatting off, --format 1`] = `
"
> node ./dist/index.cjs --format -i test:stub

pnpm akio
	-----
@abullard/akio
1. test:stub  — DO NOT REMOVE - stubbed npm script for snapshot tests

api
2. api:test:stubOne — DO NOT REMOVE - stubbed npm script for monorepo testing
3. api:test:stubTwo — DO NOT REMOVE - stubbed npm script for monorepo testing
4. api:test:stubThree — DO NOT REMOVE - stubbed npm script for monorepo testing
5. api:test:stubFour — DO NOT REMOVE - stubbed npm script for monorepo testing
6. api:test:stubFive — DO NOT REMOVE - stubbed npm script for monorepo testing
7. api:test:stubSix — DO NOT REMOVE - stubbed npm script for monorepo testing
8. api:test:stubSeven — DO NOT REMOVE - stubbed npm script for monorepo testing
9. api:test:stubEight — DO NOT REMOVE - stubbed npm script for monorepo testing

ui
10. ui:test:stubOne — DO NOT REMOVE - stubbed npm script for monorepo testing
11. ui:test:stubTwo — DO NOT REMOVE - stubbed npm script for monorepo testing
12. ui:test:stubThree — DO NOT REMOVE - stubbed npm script for monorepo testing
13. ui:test:stubFour — DO NOT REMOVE - stubbed npm script for monorepo testing
14. ui:test:stubFive — DO NOT REMOVE - stubbed npm script for monorepo testing
15. ui:test:stubSix — DO NOT REMOVE - stubbed npm script for monorepo testing
16. ui:test:stubSeven — DO NOT REMOVE - stubbed npm script for monorepo testing
17. ui:test:stubEight — DO NOT REMOVE - stubbed npm script for monorepo testing
"
`;

exports[`no-format2.spec.ts > should turn formatting off, --no-format 1`] = `
"
> node ./dist/index.cjs --no-format -i test:stub

pnpm akio
	-----
@abullard/akio
1. test:stub  — DO NOT REMOVE - stubbed npm script for snapshot tests

api
2. api:test:stubOne — DO NOT REMOVE - stubbed npm script for monorepo testing
3. api:test:stubTwo — DO NOT REMOVE - stubbed npm script for monorepo testing
4. api:test:stubThree — DO NOT REMOVE - stubbed npm script for monorepo testing
5. api:test:stubFour — DO NOT REMOVE - stubbed npm script for monorepo testing
6. api:test:stubFive — DO NOT REMOVE - stubbed npm script for monorepo testing
7. api:test:stubSix — DO NOT REMOVE - stubbed npm script for monorepo testing
8. api:test:stubSeven — DO NOT REMOVE - stubbed npm script for monorepo testing
9. api:test:stubEight — DO NOT REMOVE - stubbed npm script for monorepo testing

ui
10. ui:test:stubOne — DO NOT REMOVE - stubbed npm script for monorepo testing
11. ui:test:stubTwo — DO NOT REMOVE - stubbed npm script for monorepo testing
12. ui:test:stubThree — DO NOT REMOVE - stubbed npm script for monorepo testing
13. ui:test:stubFour — DO NOT REMOVE - stubbed npm script for monorepo testing
14. ui:test:stubFive — DO NOT REMOVE - stubbed npm script for monorepo testing
15. ui:test:stubSix — DO NOT REMOVE - stubbed npm script for monorepo testing
16. ui:test:stubSeven — DO NOT REMOVE - stubbed npm script for monorepo testing
17. ui:test:stubEight — DO NOT REMOVE - stubbed npm script for monorepo testing
"
`;
